source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -573,6 +579,25 @@ std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> ContextImpl::getPrivateKeyMe
   return providers;
 }
 
+std::vector<std::pair<std::string, uint64_t>>
+ContextImpl::getSecondsUntilExpirationOfCerts() const {
+  std::vector<std::pair<std::string, uint64_t>> certs_expiration_time;
+  // Add CA cert expiration time.
+  auto ca_cert_expiration = cert_validator_->getSecondsUntilExpirationOfCert();
+  if (ca_cert_expiration.has_value()) {
+    certs_expiration_time.emplace_back(ca_cert_expiration.value().first,
+                                       ca_cert_expiration.value().second);
+  }
+  // Add TLS certs' expiration time.
+  for (auto& ctx : tls_contexts_) {
```

Comment:
const auto& if possible

Reply:
Please use explicit type. It doesn't look like it can be const though.